### PR TITLE
UAF-53 changed the defaultOpen to true on VendorMaintenanceDocument.xml and added validation for the Supplier Diversity section

### PR DESCRIPTION
**Pull request by mmoen**

### CODE REVIEW

File: kfs-core/src/main/java/org/kuali/kfs/vnd/VendorPropertyConstants.java [hunk 1]
```diff
@@ -45,8 +45,8 @@
     public static final String VENDOR_ADDRESS_EMAIL = "vendorAddresses.vendorAddressEmailAddress";
     public static final String VENDOR_DEFAULT_ADDRESS_INDICATOR = "vendorDefaultAddressIndicator";
     public static final String VENDOR_DEFAULT_ADDRESS_CAMPUS = "vendorCampusCode";
-    public static final String VENDOR_SUPPLIER_DIVERSITY_CODE = "vendorSupplierDiversities.vendorSupplierDiversityCode";
```
Comment (ghost) [inline kfs-core/src/main/java/org/kuali/kfs/vnd/VendorPropertyConstants.java:48]: Add new constants as needed. Please do not delete delivered constants.

File: kfs-core/src/main/java/org/kuali/kfs/vnd/document/validation/impl/VendorRule.java [hunk 2]
```diff
@@ -82,6 +84,9 @@
 
     private VendorDetail oldVendor;
     private VendorDetail newVendor;
+    
+    private static final String VENDOR_SUPPLIER_DIVERSITY_CODE = "vendorSupplierDiversityCode";
```
Comment (ghost) [inline kfs-core/src/main/java/org/kuali/kfs/vnd/document/validation/impl/VendorRule.java]: Please avoid "magic strings". Constants (from a constants file) should be used instead.

File: kfs-core/src/main/java/org/kuali/kfs/vnd/document/validation/impl/VendorRule.java [hunk 3]
```diff
@@ -233,12 +238,48 @@ private boolean processValidation(MaintenanceDocument document) {
...
+    	if(vendorSupplierDiversities == null || vendorSupplierDiversities.isEmpty()) {
+    		valid = false;
+    		String fieldName = VENDOR_SUPPLIER_DIVERSITY_CODE;
+    		String attributeLabel = dataDictionaryService.getAttributeLabel(VendorSupplierDiversity.class, fieldName);
+    		String shortLabel = dataDictionaryService.getAttributeShortLabel(VendorSupplierDiversity.class, fieldName);
+    		
+    		putFieldError(VendorPropertyConstants.VENDOR_SUPPLIER_DIVERSITY_CODE, RiceKeyConstants.ERROR_REQUIRED, attributeLabel + " (" + shortLabel + ")");
+    	}
+    	
+    	else {
```
Comment (Perrinek) [inline kfs-core/src/main/java/org/kuali/kfs/vnd/document/validation/impl/VendorRule.java]: This validation for active diversity information was not included in the functional spec. It also should not be needed. The available list of supplier diversities should only include active diversity values.

**Comment by ghost** [comment]
Please see specific notes. Also, please move this modification to edu.arizona. As we are not currently contributing, we are not modifying delivered code (where at all possible).

**Comment by ghost** [comment]
Looks good! Just had a MAM with Heather/Misty/Gail and they agree this is easily testable before the next release. Will merge.

**Pull request merged**